src/main/java/com/metamx/emitter/core/factory/EmitterFactory.java

```diff
@@ -0,0 +1,11 @@
+package com.metamx.emitter.core.factory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.metamx.common.lifecycle.Lifecycle;
+import com.metamx.emitter.core.Emitter;
+import com.metamx.http.client.HttpClient;
+
+public interface EmitterFactory
```

Comment:
I took a look how it can be with Factory interface annotated, and it actually doesn't make it easier if we want to be backward compatible (old configs for http and logging emitters should work) and also to support custom emitter factory.